### DISCUSSION TITLE
response cookie values need use 'utf-8' encode

### DISCUSSION THREAD

**Discussion by Kewings** [fastapi/fastapi#6258] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
response.set_cookie(key=k, value=v, max_age=90 * 24 * 60 * 60)
  File "/xx/.local/lib/python3.8/site-packages/starlette/responses.py", line 132, in set_cookie
    self.raw_headers.append((b"set-cookie", cookie_val.encode("latin-1")))
UnicodeEncodeError: 'latin-1' codec can't encode characters in position 18-19: ordinal not in range(256)
```

### Description

![image](https://user-images.githubusercontent.com/29881381/179967791-d4896fad-3ce6-4be0-a5fd-2704e054b123.png)
i found it had been hard code in the method

or where can i change it?

### Operating System

Linux

### FastAPI Version

0.78.0

### Python Version

3.8.10

**Comment by chbndrhnns** [comment]
Based on the stack trace, I believe that's an issue in Starlette, Not FastAPI.

**Comment by JarroVGIT** [comment]
This is not a FastAPI issue, but a Starlette implementation. 

But in regards to your question, the [RFC 7230](https://datatracker.ietf.org/doc/html/rfc7230) requires that it is encoded with latin-1, and they follow that specification. Chances are small they will change this any time soon, so either make sure your headers are latin-1 encodable or make your own version of Starlette with this change in it.

**Comment by Kewings** [comment]
got it.

many thanks @JarroVGIT @chbndrhnns
